### ISSUE TITLE
Undefined dataset is returned as 200 response

### ISSUE THREAD

**Issue by oncletom** [GFDRR/open-risk-data-dashboard#349]
When I request for `https://dev.riskopendata.org/api/dataset/?country=undefined` (not an ID) or `https://dev.riskopendata.org/api/dataset/?country=` (absent ID) or `https://dev.riskopendata.org/api/dataset/?country=9999` (unexisting ID), the API returns a 200 response.

- I expect it to return a 404 when the country does not exist in the database.
- I expect it to return a 200 with an empty array if the country exists but there is not data.
- I expect it to return a 200 with array items if the country exists and data exists

**Comment by daniviga** [comment]
This is not a bug but was by design. It's a REST API so 404 is sent back if the API url does not exist, not if the resource doesn't.

You are calling a proper endpoint so a 200 is returned. In the json you'll get the error and the reason why. It's an application error what you are looking for, not an HTTP protocol one.

**Comment by nastasi-oq** [comment]
>  * I expect it to return a 404 when the country does not exist in the database.
>  * I expect it to return a 200 with an empty array if the country exists but there is not data.

I agree with you if the API offer a specialization of the country as part of the URI, not as a filter as it is now.
If we need this feature we can think to change the API accordingly.
